### DISCUSSION TITLE
How to solve "Syntax error, unrecognized expression" error for custom commands

### DISCUSSION THREAD

**Discussion by saifulrip71** [cypress-io/cypress#29247] [Questions and Help]
Hello,

I was learning Cypress from here https://learn.cypress.io/testing-your-first-application/how-to-test-forms-and-custom-cypress-commands , I tried to do the same here for custom commands. However, when running this , I am getting this error in browser. Please see the attachments below:
![image](https://github.com/cypress-io/cypress/assets/11022572/4391ec1c-cd9b-44dc-ad5f-8dd933cca912)
![image](https://github.com/cypress-io/cypress/assets/11022572/0b8853a0-7b1d-4de2-b9e1-9fc927253ff2)
![image](https://github.com/cypress-io/cypress/assets/11022572/ace86df8-73c0-4a90-8b33-bb09c18ae039)

**Comment by portyu9** [comment]
Hi @saifulrip71 

The issue is this line:

```ts
return cy.get('[data-test=${selector}]')
```

`${selector}` is only interpolated inside a JavaScript **template literal** using backticks. With single quotes, Cypress receives the literal selector:

```text
[data-test=${selector}]
```

which is invalid CSS.

Change it to:

```ts
Cypress.Commands.add('getByData', (selector) => {
  return cy.get(`[data-test="${selector}"]`)
})
```

Then:

```ts
cy.getByData('hero-heading')
```

will correctly generate:

```css
[data-test="hero-heading"]
```

Please mark this answer as accepted if it helped, thank you.
